packages/flutter/lib/src/gestures/events.dart

```diff
@@ -288,36 +288,32 @@ abstract class PointerEvent {
   final bool synthesized;
 
   @override
-  String toString() => '$runtimeType($position)';
-
-  /// Returns a complete textual description of this event.
-  String toStringFull() {
-    return '$runtimeType('
-             'timeStamp: $timeStamp, '
-             'pointer: $pointer, '
-             'kind: $kind, '
-             'device: $device, '
-             'position: $position, '
-             'delta: $delta, '
-             'buttons: $buttons, '
-             'down: $down, '
-             'obscured: $obscured, '
-             'pressure: $pressure, '
-             'pressureMin: $pressureMin, '
-             'pressureMax: $pressureMax, '
-             'distance: $distance, '
-             'distanceMin: $distanceMin, '
-             'distanceMax: $distanceMax, '
-             'size: $size, '
-             'radiusMajor: $radiusMajor, '
-             'radiusMinor: $radiusMinor, '
-             'radiusMin: $radiusMin, '
-             'radiusMax: $radiusMax, '
-             'orientation: $orientation, '
-             'tilt: $tilt, '
-             'platformData: $platformData, '
-             'synthesized: $synthesized'
-           ')';
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(DiagnosticsProperty<Duration>('timeStamp', timeStamp, defaultValue: Duration.zero));
+    properties.add(IntProperty('pointer', pointer));
```

Comment:
we should apply a `level` that hides the ones that weren't shown in toString before

Reply:
That would be basically everything. I thought part of the idea of the change here was to align the behavior of PointerEvent's toString with other classes that use Diagnosticable?

Reply:
One option would be to give every property except `position` a level of `debug`, and then keep `toStringFull` as a call to `toString` with a level of `debug`. That does seem to go against the guidance in the `Diagnosticable` and `DiagnosticsProperty` comments, but if that's the preferred approach I can update to do that.

Reply:
> keep `toStringFull` as a call to `toString` with a level of `debug`

I ended up doing this part (although with `hidden` rather than `debug` to actually print everything, the way toStringFull did before) in the interest of minimizing breaking changes.

Let me know if you still want me to change the levels on everything other than `position` though.

Reply:
I would recommend using`DiagnosticLevel.fine` instead of `DiagnosticLevel.hidden` for toStringFull. That should have the same effect and be a little more idiomatic.

Reply:
Done, thanks for the tip!

Do you have any thoughts on Hixie's suggestion of assigning a higher level to all but one of the properties here? It sounded to me like the docs were discouraging something like that, but I don't have much context so I may well have misunderstood.

Reply:
> Done, thanks for the tip!
> 
> Do you have any thoughts on Hixie's suggestion of assigning a higher level to all but one of the properties here? It sounded to me like the docs were discouraging something like that, but I don't have much context so I may well have misunderstood.

Where in the docs is it discouraging that? I want to update the docs if that is the case. There are existing cases where we do just that and I expect there will be more cases once the lints to warn about missing properties land.

Using a lower level to filter out content that should not be shown in the shorter toString() but should be shown in other debugging tools is a good way to use DiagnosticsNode objects. You will see some existing cases in the code base where properties that we want to show in the inspector but don't want to show in the toStringDeep output are filtered that way.

Reply:
Hm, this turned out to be partially user error. I somehow had an off-by-one error in the mapping of what levels mean to the values and the defaults, and would need to be marking everything as being what the levels docs described as "low value".

The main thing I was thinking of other than that was:
```
Avoid specifying the `level` parameter unless the result you want
cannot be achieved by using the `defaultValue` parameter or using
the [ObjectFlagProperty] class to conditionally display the property
as a flag.
```
It sounds like I over-anchored on the part about avoiding using it. I read this to indicate that using it should be rare, and thus applying it to >90% of the properties on an object felt strange.

And to a small extent, the fact that `Diagnosticable` has a `toStringShort`, whose comment seems to match what the existing `PointerEvent.toString` was doing, suggested to me that `toString` wasn't intended to be such a terse description in a `Diagnosticable` implementation.


It sounds like I should just make almost everything debug-level to make this change as low-impact as possible, per Hixie's suggest, and then leave it to others to tweak the display if they want to.